esp/scm/ws_config2.ecm

```diff
@@ -0,0 +1,368 @@
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+
+ESPstruct statusMsgType
+{
+    string msgLevel;                // message level ("info", "warning", "error", "fatal")
+    string nodeId;                  // node identifier that generated this status message
+    string attribute("");           // name of node's attribute generating the message ("" for the node itself)
+    string msg;                     // the message
+    ESParray<string, parentId> parentIdList;
+};
+
+ESPstruct statusType
+{
+    bool error(false);    // true if a message exsits in status (below) with a message level of error or higher
+    ESParray<ESPstruct statusMsgType, statusMsg> status;
+};
+
+
+ESPresponse PassFailResponse
+{
+    bool error(false);
+    string msg("");
+};
+
+
+ESPresponse PassFailWithStatusResponse
+{
+    bool   error(false);
+    string msg("");
+    ESPstruct statusType status;
+};
+
+
+//
+// Session management (not to be confused with platform session management)
+
+ESPrequest OpenSessionRequest
+{
+    string username;
+    string type("XML");           // type of session
+    string schemaPath("");        // location of configuration schema environmentFiles
+    string masterSchemaFile("");  // name of master schema control file
+    string sourcePath("");        // path to envrionment files
+    string activePath("");        // path to active environment file ("" for none)
+};
+
+
+ESPresponse OpenSessionResponse
+{
+    bool   error(false);
+    string sessionId;
+    string msg("");         // any messages from the open session request
+};
+
+
+ESPrequest CloseSessionRequest
+{
+    string sessionId;
+    bool forceClose(false);    // must be true to close a session that has a modified environmrnt file
+};
+
+
+ESPrequest ListOpenSessionsRequest
+{
+    string dummyParam("");    // not used
+};
+
+
+ESPstruct openSessionInfo
+{
+    string sessionId;
+    string username;
+    string curEnvironmentFile("");
+    bool   locked;
+    bool   modified;
+};
+
+
+ESPresponse ListOpenSessionsResponse
+{
+    bool   error(false);
+    string msg("");
+    ESParray<ESPstruct openSessionInfo, sessionInfo> openSessions;
+};
+
+
+//
+// Node management
+
+ESPrequest NodeRequest
+{
+    string sessionId;
+    string nodeId;
+};
+
+
+ESPrequest RemoveNodeRequest
+{
+    string sessionId;
+    string sessionLockKey("");
+    string nodeId;
+};
+
+
+ESPstruct choiceType
+{
+    string displayName;
+    string value;
+    string desc("");
+};
+
+
+ESPstruct limitsType
+{
+    bool   minValid(false);
+    bool   maxValid(false);
+    int    min(0);
+    int    max(0);
+    ESParray<ESPstruct choiceType, choice> choiceList;
+    //ESParray<string, disallow> disallowList;
+    ESParray<string, expr> regex;
+};
+
+
+ESPstruct typeInfo
+{
+    string name("na");
```

Comment:
lol == laugh out loud ;)
ok, so I'm not sure if "Not Available" is better than just leaving it empty...
In any case it's prob clearer to send back "Not Available" rather than "na"